### PR TITLE
WIP: Active directory support V1

### PR DESCRIPTION
**Pull request by MalloZup**
# description:

add a V1 initial version of active directory.  As you can see the branch is ```SUSE:develop-active-directory``` so we will merge it later on `develop` branch once we get feedback.

Unfortunately,  I couldn't test the azure part

### CODE REVIEW

**Comment by MalloZup** [comment]
image

**Comment by MalloZup** [comment]
why this is closed?

**Comment by stefanotorresi** [comment]
my mistake! sorry!

File: libvirt/variables.tf [hunk 1]
```diff
@@ -47,6 +47,30 @@ variable "public_key_location" {
...
+  type        = string
+  default     = ""
+}
+
+variable "ad_server_domain" {
+  description = "name of the AD domain we want to join"
+  type        = string
+  default     = ""
+}
+
```
Comment (arbulu89) [inline libvirt/variables.tf:69]: Remove all of these new lines please.

File: libvirt/variables.tf [hunk 2]
```diff
@@ -47,6 +47,30 @@ variable "public_key_location" {
...
+# active directory server variables
+
+variable "ad_server" {
+  description = "IP adress of active directory server"
+  type        = string
+  default     = ""
+}
+
+variable "ad_adm_pwd" {
+  description = "password for admin of domain"
```
Comment (arbulu89) [inline libvirt/variables.tf:59]: The rest of the descriptions are started with capital letter. This should follow the rule

**Review by arbulu89 — CHANGES_REQUESTED** [review]
@MalloZup In general it looks a good implementation. I added some small comments. And besides them:

We will need to figure it out how to deal with different role nodes (hana, netweaver, monitoring). I would assume that someone who is using AD, would like to have all the machines associated to this AD server, but obviously, not all the users should belong to all machines. Do we know how to do so?

**Comment by MalloZup** [comment]
I think if the admin wants to have granularity and fine acess control, 
for netweaver/etc, the solution can be that the AD admin create subdomain

`foo.test.hana`, `foo.test.netweaver` `foo.test.monitoring`. etc

i think we could create like a list of map in terraform where the user specify host/role to AD

in pseudo code
`hana: foo.test.hana`, `netweaver: foo.test.netweaver` etc
`hana_pdw` 
and by default if there is no map but a global domain, just join all machine to this

somehting in this direction

File: salt/active_directory/setup.sls [hunk 3]
```diff
@@ -0,0 +1,114 @@
+install_sssd_packages:
+  pkg.latest:
+    - pkgs:
+      - realmd
+      - adcli
+      - sssd
+      - sssd-tools
+      - sssd-ad
+      - samba-client
+      - tcsh
```
Comment (MalloZup) [inline salt/active_directory/setup.sls:10]: we don't need this will remove

File: libvirt/variables.tf (hunk 2 above)
Comment (MalloZup) [reply libvirt/variables.tf:59]: ok

**Comment by MalloZup** [comment]
```module.netweaver_node.module.netweaver_provision.null_resource.provision[2] (remote-exec): [ERROR   ] Command '['groupmod', '-g', 2002, 'sapsys']' failed with return code: 10
module.netweaver_node.module.netweaver_provision.null_resource.provision[2] (remote-exec): [ERROR   ] stdout: groupmod: group 'sapsys' does not exist in /etc/group
module.netweaver_node.module.netweaver_provision.null_resource.provision[2] (remote-exec): [ERROR   ] retcode: 10
``` 
seems in netweaver we hardcoded groups with `groupmod`

**Comment by stefanotorresi** [comment]
> seems in netweaver we hardcoded groups with `groupmod`

what salt state was this in?

**Comment by MalloZup** [comment]
@stefanotorresi  it is an upstream call group.present which we call on netweaver formula.

However due the early stage of the `netweaver` AD implementation/integration I need to check if it is the culprit or is an `error` we have just because of that.

**Comment by MalloZup** [comment]
Closing this since we will just offer some documentation which I'm working with thomas. is ok to close

**Pull request closed without merging**